src/detect-engine-threshold.c

```diff
@@ -300,11 +300,11 @@ static inline void RateFilterSetAction(Packet *p, PacketAlert *pa, uint8_t new_a
             pa->flags |= PACKET_ALERT_RATE_FILTER_MODIFIED;
             break;
         case TH_ACTION_DROP:
-            PacketDrop(p, PKT_DROP_REASON_RULES_THRESHOLD);
+            PacketDrop(p, ACTION_DROP, PKT_DROP_REASON_RULES_THRESHOLD);
             pa->flags |= PACKET_ALERT_RATE_FILTER_MODIFIED;
             break;
         case TH_ACTION_REJECT:
-            PACKET_REJECT(p);
```

Comment:
With your changes, `PACKET_REJECT` is no longer used so I suggest removing
- PACKET_REJECT
- PACKET_REJECT_BOTH
- PACKET_REJECT_DST

Related, `PacketSetAction` is no longer used -- suggest removal.

Reply:
in a simple case like this I prefer removing an unused macro or function right way

Reply:
Ok, will do, then. But I still think that decode.h has more work to be done around that. I'll bring that topic up again, so we won't forget about it.

Reply:
:tickets: :)

Reply:
https://redmine.openinfosecfoundation.org/issues/5517

:P